main.tf

```diff
@@ -46,6 +46,7 @@ module "hush_sensor" {
   service_name           = var.sensor_service_name
   cluster_name           = var.cluster_name
   execution_role_arn     = aws_iam_role.hush_ecs_role.arn
+  task_role_arn          = aws_iam_role.hush_ecs_role.arn
```

Comment:
i dont exactly understand why the task needs a role, and if this is part of this PR.
lets better understand what going on here: maybe the task needs a different role than the execution_role_arn

Reply:
@zilbers hoo i think i know what happens:

you have a SINGLE role with 2 different permissions: get the image from ECR; and get our deploiyment secrets from SM.
the one is needed by the execution system; the second is needed by our task software.

formerly, you placed the role into execution_role_arn - since both the executor (host) AND the task software are RUNNING AS THE HOST.

after the "awsvpc" change, the task software is NO LONGER RUNNING AS THE HOST, so fetching the secret needs an task role.

What i would do is split this to 2 different IAM roles, one for execution (ecr) and one for the task (sm).